src/uniprotkb/components/entry/tabs/Publications.tsx

```diff
@@ -24,22 +24,110 @@ import { processUrlTemplate } from '../../../../shared/utils/xrefs';
 import {
   CitationsAPIModel,
   Reference,
+  Source,
 } from '../../../../supporting-data/citations/adapters/citationsConverter';
 import LiteratureCitation from '../../../../supporting-data/citations/components/LiteratureCitation';
 import apiUrls from '../../../config/apiUrls/apiUrls';
 import { getParamsFromURL } from '../../../utils/resultsUtils';
 
+const orcidIDRegExp = /(\d{4}-){3}\d{4}/;
+
+type PublicationSourceProps = {
+  accession: string;
+  source: Source;
+};
+
+export const PublicationSource = ({
+  accession,
+  source,
+}: PublicationSourceProps) => {
+  const databaseInfoMaps = useDatabaseInfoMaps();
+
+  if (source.name.includes('UniProtKB')) {
+    return (
+      <>
+        <EntryTypeIcon entryType={source.name} />
+        {source.name}
+      </>
+    );
+  }
+
+  let uriLink =
+    source && databaseInfoMaps?.databaseToDatabaseInfo[source.name]?.uriLink;
+
+  if (source.name === 'ORCID') {
+    return (
+      <>
+        <EntryTypeIcon entryType={source.name} />
+        {source.name}:{' '}
+        <ExternalLink
+          url={
+            source.id && orcidIDRegExp.test(source.id)
+              ? processUrlTemplate(uriLink, { id: source.id })
+              : null
+          }
+        >
+          {source.id}
+        </ExternalLink>
+        {' ('}
+        <ExternalLink
+          url={`//community.uniprot.org/bbsub/bbsubinfo.html?accession=${accession}`}
+        >
+          see community submission
+        </ExternalLink>
+        ).
+      </>
+    );
+  }
+
+  /* URI link exceptions: */
+  // 'GeneRif' is indexed as 'GeneRIF' in the configuration endpoint
+  if (source.name === 'GeneRif') {
```

Comment:
Is there a misconfiguration somewhere along the way (production/backend) which has caused this case mismatch? Is the proposed code change a temporary fix or a more permanent edge case handler?